src/feature_frame_buffer.c

```diff
@@ -133,26 +133,28 @@ static void update_proc_frame_buffer(Layer *layer, GContext *ctx) {
 
   // obtain frame buffer (must be released again using graphics_release_frame_buffer!)
   GBitmap *fb = graphics_capture_frame_buffer(ctx);
+  GRect fb_bounds = gbitmap_get_bounds(fb);
 
   // during the loop, row always points to the byte of the first 8 pixels of the current row
-#ifdef PBL_PLATFORM_BASALT
-  GRect fb_bounds = gbitmap_get_bounds(fb);
+#ifdef PBL_ROUND
+  GBitmapDataRowInfo row_info;
+#else
   uint8_t *row = gbitmap_get_data(fb);
   uint16_t row_bytes = gbitmap_get_bytes_per_row(fb);
-#else
-  GRect fb_bounds = fb->bounds;
-  uint8_t *row = fb->addr;
-  uint16_t row_bytes = fb->row_size_bytes;
-#endif
   row += fb_bounds.origin.y * row_bytes;
+#endif
+  
 
   // For all rows
   for (int16_t y = fb_bounds.origin.y; y < fb_bounds.size.h; y++) {
   const uint8_t row_gray = gray_for_row(y);
-#ifdef PBL_PLATFORM_BASALT
+#ifdef PBL_ROUND
+      row_info = gbitmap_get_data_row_info(fb, y);
+#endif
+#ifdef PBL_SDK_3
     // Each pixel is one whole byte
-    for(int x = fb_bounds.origin.x; x < fb_bounds.size.w; x++) {
-      memset(&row[(y * fb_bounds.size.w) + x], color_for_gray(x, y, row_gray).argb, 1);
+    for(int x = PBL_IF_RECT_ELSE(fb_bounds.origin.x, row_info.min_x); x < PBL_IF_RECT_ELSE(fb_bounds.size.w, row_info.max_x + 1); x++) {
```

Comment:
Use an intermediary variable (such as `min_x`) and set it to the appropriate value instead of these very long lines.